VideoLocker/src/main/java/org/edx/mobile/model/course/BlockType.java

```diff
@@ -4,6 +4,12 @@
  * Created by hanning on 5/21/15.
  */
 public enum  BlockType {
-    COURSE, CHAPTER,  SECTION,  SEQUENTIAL,
-    VERTICAL,  VIDEO,  HTML, PROBLEM, DISCUSSION, OTHERS
+    COURSE(false), CHAPTER(false),  SECTION(false),  SEQUENTIAL(false),
```

Comment:
> I hadn't realized Java doesn't properly deal with enum exhaustivity - but I guess it's necessary since `null` is a valid input.

Actually, `null` is not a valid case when switching on an enum; that would just throw a `NullPointerException` instead. There is no obvious reason why Java does not recognize the handling of all `enum` cases. There are a few speculations on this subject at https://stackoverflow.com/questions/5013194/why-is-default-required-for-a-switch-on-an-enum-in-this-code

> I guess if we do add more fields, we can factor this to be methods on an abstract type or something like that, so we don't just end up with a bunch of constructors with unnamed properties that are hard to read at a glance.

In that case we could just add a comment specifying the arguments before defining the items, if readability is the goal. If there are too many arguments then it can potentially be formatted as a table.